source/braille.py

```diff
@@ -1441,11 +1441,150 @@ def _setCursor(self, info: textInfos.TextInfo):
 class ReviewTextInfoRegion(TextInfoRegion):
 
 	allowPageTurns=False
+	_currentSelection: textInfos.TextInfo | None = None
+	_currentScriptName: str | None = None
+	_withinSelection: bool = False
+
+	def _getSelection(self) -> textInfos.TextInfo:
+		"""Gets selection for use in update function.
+		:return: selection, reading unit from it, or review position when
+		there is no selection or showing selection is disabled
+		"""
+		self._withinSelection = False
+		if not config.conf["braille"]["showSelection"]:
+			self._currentSelection = None
+			return api.getReviewPosition().copy()
+		info: textInfos.TextInfo
+		try:
+			info = self.obj.makeTextInfo(textInfos.POSITION_SELECTION)
+		except (LookupError, RuntimeError):
+			self._currentSelection = None
+			info = api.getReviewPosition().copy()
+			# Info should be collapsed, but it is not always, at least when
+			# switching from focus mode to browse mode.
+			info.collapse()
+			return info
+		if info.isCollapsed:
+			# Cursor
+			self._currentSelection = None
+			info = api.getReviewPosition().copy()
+			# Info should be collapsed, but it is not always, at least when
+			# switching from focus mode to browse mode.
+			info.collapse()
+			return info
+		if (
+			self._currentSelection is None
+			or self._currentSelection.start != info.start
+			or self._currentSelection.end != info.end
+		):
+			# Selection changed, update also review position
+			self._currentSelection = info.copy()
+			if self.obj.isTextSelectionAnchoredAtStart:
+				self._currentSelection.move(textInfos.UNIT_CHARACTER, -1, "end")
+			self._currentSelection.collapse(end=self.obj.isTextSelectionAnchoredAtStart)
+			# Enqueue to avoid recursion error
+			queueHandler.queueFunction(queueHandler.eventQueue, api.setReviewPosition, self._currentSelection)

```

Comment:
I think updating the review position from a _getSelection function should be avoided. Again, please explain the intention of the code.

Reply:
Updating review position when selection changes and braille is tethered to review (at least when review follows caret) is important so that selection changes are immediately shown in braille.

Reply:
Please explain in code as requested.